File: app/src/main/java/com/pgssoft/mvvm/services/ServiceProvider.java
package com.pgssoft.mvvm.services;

import android.content.Context;

import com.pgssoft.mvvm.services.api.RetrofitService;
import com.pgssoft.mvvm.services.database.DBHandler;
import com.pgssoft.mvvm.services.interfaces.ApiService;
import com.pgssoft.mvvm.services.interfaces.MapperService;
import com.pgssoft.mvvm.services.interfaces.Repository;
import com.pgssoft.mvvm.services.mock.MockRepository;
import com.pgssoft.mvvm.services.mock.ObjectMapper;

/**
 * Created by bstokrocki on 29.01.2017.
 */
public class ServiceProvider {
    private ApiService apiService;
    private Repository repository;
    private MapperService mapperService;

    public ServiceProvider(Context applicationContext) {
        mapperService = new ObjectMapper();
        apiService = new RetrofitService();
        repository = new DBHandler(applicationContext, mapperService);
    }

    public ApiService getApiService() {
        return apiService;
    }

    public Repository getRepository() {
        return repository;
    }

    public MapperService getMapperService() {
        return mapperService;
    }
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/service/ServiceProvider.kt
package com.pgssoft.kotlinplayground.service

import android.content.Context
import com.nostra13.universalimageloader.core.ImageLoader
import com.nostra13.universalimageloader.core.ImageLoaderConfiguration
import com.pgssoft.kotlinplayground.service.api.RetrofitService
import com.pgssoft.kotlinplayground.service.db.DBHandler
import com.pgssoft.kotlinplayground.service.db.DBHelper
import com.pgssoft.kotlinplayground.service.mock.ObjectMapper
import net.danlew.android.joda.JodaTimeAndroid

/**
 * Created by bstokrocki on 13-Apr-17.
 */
class ServiceProvider(applicationContext: Context) {
    val scheduler = SchedulerService()
    val mapper = ObjectMapper()
    val repository = DBHandler(DBHelper(applicationContext), mapper)
    val apiService = RetrofitService()

    init {
        JodaTimeAndroid.init(applicationContext)
        ImageLoader.getInstance().init(ImageLoaderConfiguration.createDefault(applicationContext))
    }
}